src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2677,7 +2677,7 @@ TransactionStateMachine(WorkerSession *session)
 				}
 				else
 				{
-					transaction->transactionState = REMOTE_TRANS_INVALID;
```

Comment:
We should watch out for this one. We could (soon) get into situations where we're resetting the transaction state on a session that's actually in a transaction block.

Reply:
Fixed in https://github.com/citusdata/citus/pull/3288/files